config/formats/typescript-export-defition.ts

```diff
@@ -0,0 +1,145 @@
+
+import StyleDictionary from 'style-dictionary';
+import { format } from "prettier";
+const fs = require('fs');
+const path = require('path');
+
+const { fileHeader } = StyleDictionary.formatHelpers;
+
+/**
+ * unquoteTypes
+ * @description extract types from designTokenTypes (string with type definitions from file) and replace quoted types "Color" with unquoted Color
+ * @param output 
+ * @param designTokenTypes 
+ * @returns string with unquoted types
+ */
+const unquoteTypes = (output: string, designTokenTypes: string[]): string => {
+  // regex to match types like `type Color`
+  // const findTypesRegex = `type ([A-Za-z]+)`
+  // // extract types from designToken
+  // const types = [...designTokenTypes.matchAll(new RegExp(findTypesRegex, "g"))].map(match => match[1])
+  // join types for replacement
+  const regex = `"(${designTokenTypes.join('|')})"`
+  // remove strings
+  return output.replace(new RegExp(regex, "g"), '$1')
+}
+
+/**
+ * getTokenType
+ * @description extract content from token files
+ * @param path string
+ */
+const getTokenType = (tokenTypesPath: string): string => {
+  const designTokenType = fs.readFileSync(
+    path.resolve(tokenTypesPath), { encoding: 'UTF-8' }
+  );
+  //
+  return designTokenType
+}
+
+/**
+ * convertPropToType
+ * @description return take type attribute and return correct typescript type as string 
+ * @param type 
+ * @returns 
+ */
+const convertPropToType = (value: any, type: string): string => {
+  switch (type) {
+    case 'color':
+      if (value[0] === '#') {
+        return 'ColorHex';
+      }
+    case 'dimension':
+      if (value.substring(value.length - 3) === 'rem') {
+        return 'SizeRem';
+      }
+      return 'Size'
+    case 'shadow':
+      return 'Shadow';
+    default:
+      return 'string';
+  }
+}
+
+/**
+ * toType
+ * @description converts tokens to type and returns array with used type
+ * @param token StyleDictionary.DesignToken in object structure to their types using the $type attribute
+ * @returns 
+ */
+const toType = (token: StyleDictionary.DesignToken | {}, usedTypes: Set<string>): object | string => {
+  // is design token
+  if ('value' in token) {
+    usedTypes.add(convertPropToType(token.value, token.$type))
+    return convertPropToType(token.value, token.$type)
+  }
+  // is non-object value
+  if (typeof token !== 'object') return token
+  // is obj
+  const nextObj = {}
+  for (const [prop, value] of Object.entries(token)) {
+    // @ts-ignore
```

Comment:
This was just too much for me 😅 

> config/formats/typescript-export-defition.ts:77:5 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
>   No index signature with a parameter of type 'string' was found on type '{}'.